cmd/exit.go

```diff
@@ -14,10 +14,14 @@ const (
 	// 5: CodeNotFound
 	CodeFailedDetect           = 100
 	CodeFailedDetectWithErrors = 101
-	CodeFailedBuild            = 7
+	CodeDetectError            = 102
+	CodeAnalyzeError           = 200
+	CodeRestoreError           = 300
+	CodeFailedBuild            = 400
```

Comment:
Here I think we need a generic `CodeBuildError` and then different `CodeFailedBuildWithError` (or something like that), set to `401` for buildpack error:

https://github.com/buildpacks/spec/blob/platform/0.4/platform.md#outputs-3